js/interface.js

```diff
@@ -0,0 +1,64 @@
+var appDataSources = [];
+Fliplet.DataSources.get({ appId: Fliplet.Env.get('masterAppId'), attributes: ['id', 'name'] }).then(function (DSs) {
+  appDataSources = DSs.map(el => {
+    return { value: el.id, label: el.name };
+  });
+  Fliplet.Widget.generateInterface({
+    title: 'File list',
+    fields: [
+      {
+        name: 'dataSource',
+        type: 'dropdown',
+        label: 'Data Source',
+        options: appDataSources,
```

Comment:
If you're using a dropdown to ask users to pick a data source, shouldn't that be using the data source provider?

Reply:
Not sure I follow? Can you please clarify? Thanks @tonytlwu